src/euler_characteristic.cpp

```diff
@@ -1,8 +1,29 @@
+#include "edges.h"
 #include "euler_characteristic.h"
+#include <vector>
+#include <algorithm>
 
 int euler_characteristic(const Eigen::MatrixXi &F)
 {
-  int Chi;
-  // ADD YOUR CODE HERE
+  // Chi = |V| - |E| + |F|
+  int Chi, v1, v2;
+  Eigen::MatrixXi E = edges(F);
+  
+  // Find |V|
+  std::vector<int> vertices;
+  for (int i = 0; i < E.rows(); i = i + 1) {
```

Comment:
This loop is O(n^2). Finding the number of vertices should be O(n log n) in the worst case and O(n) if we assume that F indexes all vertices.